### PR TITLE
UI for Edit Profile Tab

### PR DESCRIPTION
**Pull request by jb162**
Created UI for EditProfileTab component within EditProfile. Allows user to input profile information into form, then displays info as an alert when "save" button clicked.

### What this PR does (required):
- displays form for the user to enter profile information
- user is either 'client' (the user that pays for dog-walking service) or 'partner' (the user that walks dogs), specified with accountType state variable
- accountType conditionally renders form components (e.g. 'client' does not have 'I'm available' or 'availability options
- upon 'save' button click, the user's input is displayed as an alert

### Screenshots / Videos (front-end only):
Screenshot

### Any information needed to test this feature (required):
- run client in terminal
```
npm run
```
- navigate to 'http://localhost:3000/edit-profile'
- input data into the form (automatically set to partnerForm, but can be changed to clientForm via accountType in EditProfileTab.tsx)
- click the 'save' button

### Any issues with the current functionality (optional):
- I created an input that replaces the text 'No Phone number entered' when the 'Add a phone number' button is clicked. This input is used for the user to enter their phone number. I followed the mockups but was unsure how exactly the 'Add a phone number' button should function. If there is a better way to implement its functionality, I would appreciate your feedback.

### CODE REVIEW

**Review by ezeilo-su — COMMENTED** [review]
Hey @jb162, your UI looks good. You should also link your ticket anytime you create a new PR.
:+1:

File: client/src/components/EditProfile/EditMenu.tsx [hunk 1]
```diff
@@ -68,7 +70,7 @@ const EditMenu = (): JSX.Element => {
       </Grid>
       <Grid item xs={6} md={6} lg={6} component={Paper} className={classes.tabComponents}>
         <TabPanel value={value} index={0}>
-          Edit Profile Placeholder
+          <EditProfileTab></EditProfileTab>
```
Comment (rajivtitus) [inline client/src/components/EditProfile/EditMenu.tsx:73]:
```suggestion
          <EditProfileTab />
```
Nitpick!

File: client/src/components/EditProfile/EditProfileTab/EditProfileTab.tsx [hunk 2]
```diff
@@ -0,0 +1,320 @@
+import useStyles from './useStyles';
+import Typography from '@material-ui/core/Typography';
+import { FormLabel, OutlinedInput, Select, MenuItem, TextField, Button, Box, Menu, Switch } from '@material-ui/core';
+import React, { useState } from 'react';
+
+interface UserData {
+  firstName: string;
+  lastName: string;
+  gender: any;
```
Comment (rajivtitus) [inline client/src/components/EditProfile/EditProfileTab/EditProfileTab.tsx:9]: Please avoid using 'any' as a type. Ditto throughout!

File: client/src/components/EditProfile/EditProfileTab/EditProfileTab.tsx [hunk 3]
```diff
@@ -0,0 +1,320 @@
+import useStyles from './useStyles';
+import Typography from '@material-ui/core/Typography';
+import { FormLabel, OutlinedInput, Select, MenuItem, TextField, Button, Box, Menu, Switch } from '@material-ui/core';
+import React, { useState } from 'react';
+
+interface UserData {
```
Comment (rajivtitus) [inline client/src/components/EditProfile/EditProfileTab/EditProfileTab.tsx:6]:
```suggestion
interface UserProfile {
```
Won't UserProfile be a more suitable name for this interface? :)

File: client/src/components/EditProfile/EditProfileTab/EditProfileTab.tsx [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+        address: ${address}\n
+        description: ${description}
+      `);
+    }
+  };
+
+  return (
+    <Box>
+      <Typography className={classes.title}>Edit Profile</Typography>
+      <form className={classes.form} onSubmit={handleSubmit}>
```
Comment (rajivtitus) [inline client/src/components/EditProfile/EditProfileTab/EditProfileTab.tsx:89]: Suggestion: Please use Formik for state management and form validation :)

File: client/src/components/EditProfile/EditProfileTab/EditProfileTab.tsx [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+            value={gender}
+            onChange={({ target: { value } }) => setGender(value)}
+            variant="outlined"
+            style={gender !== 'gender' ? { color: 'black' } : { color: 'grey' }}
+          >
+            <MenuItem value="gender" disabled>
+              gender
+            </MenuItem>
+            <MenuItem value="male">male</MenuItem>
+            <MenuItem value="female">female</MenuItem>
```
Comment (rajivtitus) [inline client/src/components/EditProfile/EditProfileTab/EditProfileTab.tsx:174]: Optional: Maybe add an option for 'non-binary' too

File: client/src/components/EditProfile/EditProfileTab/useStyles.ts [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    alignItems: 'center',
+    marginBottom: 20,
+    marginLeft: '5vw',
+    minHeight: '4.2vh',
+  },
+  label: {
+    fontSize: 14,
+    color: 'black',
+    fontWeight: 'bold',
+    textTransform: 'uppercase',
```
Comment (rajivtitus) [inline client/src/components/EditProfile/EditProfileTab/useStyles.ts:31]: 💯

File: client/src/components/EditProfile/EditProfileTab/EditProfileTab.tsx [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+            id="firstName"
+            placeholder="John"
+            name="firstName"
+            autoComplete="given-name"
+            autoFocus={accountType === 'client'}
+            value={firstName}
+            onChange={({ target: { value } }) => setFirstName(value)}
+          ></OutlinedInput>
+        </Box>
+        <Box className={classes.section}>
```
Comment (rajivtitus) [inline client/src/components/EditProfile/EditProfileTab/EditProfileTab.tsx:142]: Optional: I think we can componentize this block of code for increased readability and reusability :)

**Comment by rajivtitus** [comment]
Fantastic work with the UI 🚀 

Just some comments from my end, please have a look andaddresss 😄

**Comment by rajivtitus** [comment]
Accidentally closed this PR 😵‍💫

**Comment by rajivtitus** [comment]
Fantastic work with the UI 🚀

Just some comments from my end, please have a look and address 😄

File: client/src/components/EditProfile/EditProfileTab/EditProfileTab.tsx [hunk 8]
```diff
@@ -0,0 +1,340 @@
...
+  'October',
+  'November',
+  'December',
+];
+const days = [...Array(30).keys()].map((i) => i + 1);
+const years = [...Array(119).keys()].map((i) => i + 1903).sort((a, b) => b - a);
+
+export default function EditProfileTab(): JSX.Element {
+  const classes = useStyles();
+  const [accountType, setAccountType] = useState<UserProfile['accountType'] | null>('partner');
```
Comment (rajivtitus) [inline client/src/components/EditProfile/EditProfileTab/EditProfileTab.tsx:42]: Please note that eventually we won't need these states as we will be fetching the logged in user's profile (we can store it in a single state/in the context) 😄

**Review by rajivtitus — APPROVED** [review]
Thanks for making requested changes :)

**Pull request merged**